framework/PostProcessors/BasicStatistics.py

```diff
@@ -174,50 +172,29 @@ def inputToInternal(self, currentInp):
-    else:
+
+    metadata =  currentInput.getMeta(pointwise=True)
+    inputList = []
+    if currentInput.type == 'PointSet':
+      inputDict = {}
+      inputDict['targets'] = currentInput.getVarValues(self.parameters['targets'])
+      inputDict['metadata'] =  metadata
+      inputList.append(inputDict)
+    elif currentInput.type == 'HistorySet':
       if self.pivotParameter is None:
         self.raiseAnError(IOError, self, 'Time-dependent statistics is requested (HistorySet) but no pivotParameter got inputted!')
-      inputs, outputs  = currentInput.getParametersValues('inputs',nodeId = 'ending'), currentInput.getParametersValues('outputs',nodeId = 'ending')
-      nTs, self.dynamic = len(outputs.values()[0].values()[0]), True
-      if self.pivotParameter not in currentInput.getParaKeys('output'):
-        self.raiseAnError(IOError, self, 'Pivot parameter ' + self.pivotParameter + ' has not been found in output space of data object '+currentInput.name)
-      pivotParameter =  six.next(six.itervalues(outputs))[self.pivotParameter]
-      self.raiseAMessage("Starting recasting data for time-dependent statistics")
-      targetInput  = []
-      targetOutput = []
-      for targetP in self.parameters['targets']:
-        if targetP in currentInput.getParaKeys('output'):
-          targetOutput.append(targetP)
-        elif targetP in currentInput.getParaKeys('input'):
-          targetInput.append(targetP)
-        else:
-          self.raiseAnError(IOError, self, 'Target ' + targetP + ' has not been found in data object '+currentInput.name)
-      inputDict = {}
-      inputDict['timeDepData'] = OrderedDict((el,defaultdict(dict)) for el in pivotParameter)
-      for targetP in targetInput:
-        inputValues = np.asarray([val[targetP][-1] for val in inputs.values()])
-        for ts in range(nTs):
-          inputDict['timeDepData'][pivotParameter[ts]]['targets'][targetP] = inputValues
-      metadata = currentInput.getAllMetadata()
-      for cnt, targetP in enumerate(targetOutput):
-        outputValues = np.asarray([val[targetP] for val in outputs.values()])
-        if len(outputValues.shape) != 2:
-          self.raiseAnError(IOError, 'Histories are not syncronized! Please, pre-process the data using Interfaced PostProcessor HistorySetSync!')
-        for ts in range(nTs):
-          inputDict['timeDepData'][pivotParameter[ts]]['targets'][targetP] = outputValues[:,ts]
-          if cnt == 0:
-            inputDict['timeDepData'][pivotParameter[ts]]['metadata'] = metadata
+      self.dynamic = True
+      self.pivotValue = currentInput.asDataset()[self.pivotParameter].values
+      slices = currentInput.sliceByIndex(self.pivotParameter)
+      for sliceData in slices:
+        inputDict = {}
+        inputDict['metadata'] = metadata
+        inputDict['targets'] = dict((target, sliceData[target]) for target in self.parameters['targets'])
```

Comment:
There may be "NaN" in sliceData[target] if the histories are not aligned.  Calling "dropna" can fix this.  However, we require alignment, right?  Did "checkAlignement" get called before this?

Reply:
fixed.

Reply:
I added, but the test time_dep.xml failed, which should not fail based on my understanding. Could you take a look at that @PaulTalbot-INL

Reply:
It's hard for me to take a look at when I don't know where or how you used it, or what the error was.  Can you provide some additional reporting?  The method was made specifically for your application, so it needs to work for it.

Reply:
Fixed.  Good work.